shared/validate/index.ts

```diff
@@ -0,0 +1,67 @@
+import ZSchema from 'shared/validate/z_schema';
+import Validator from 'z-schema';
+import { ResponseEntity } from 'shared/model/response';
+import { logger } from 'shared/util/logger';
+import { MESSAGE_CHANNEL } from 'shared/driver/socket/channels';
+import { ALL_SCHEMAS, MESSAGE } from 'shared/validate/schema';
+import { API_ACTION_TYPES } from 'shared/driver/socket/codes';
+import {RPC_METHODS} from 'api/middleware/rpcHolder';
+
+/**
+ * Compile all schemas for validate
+ */
+const validator: Validator = new ZSchema({});
+const isSchemasValid: boolean = validator.validateSchema(ALL_SCHEMAS);
+logger.debug('[API][SCHEMAS] VALID:', isSchemasValid);
+
+/**
+ * Decorator for validate request
+ */
+export const validate = () => {
+    return function ( target: any, propertyName: string, descriptor: PropertyDescriptor) {
+        let descriptorFn = descriptor.value || descriptor.get();
+
+        return {
+            value: (message, socket): any => {
+                let schemaID = message.code;
+                if (schemaID === API_ACTION_TYPES.CREATE_TRANSACTION &&
+                    Object.keys(message.body || [])
```

Comment:
make it more readable. extract `Object.keys(message.body || [])
                        .concat(Object.keys(message.body.data || []))
                        .includes('trs')
                )` as variable